packages/core/src/components/product/ProductGrid/ProductGrid.tsx

```diff
@@ -26,45 +27,105 @@ interface Props {
     ProductCardProps,
     'showDiscountBadge' | 'bordered' | 'taxesConfiguration'
   >
+  /**
+   * Identify the number of firstPage
+   */
+  firstPage?: number
 }
 
 function ProductGrid({
   products,
   page,
   pageSize,
   productCard: { showDiscountBadge, bordered, taxesConfiguration } = {},
+  firstPage,
 }: Props) {
   const { __experimentalProductCard: ProductCard } =
     useOverrideComponents<'ProductGallery'>()
   const aspectRatio = 1
 
+  // TODO: Check if is also isMobile
+  const isFirstPage = firstPage === page
+
   return (
     <ProductGridSkeleton
       aspectRatio={aspectRatio}
       loading={products.length === 0}
     >
       <UIProductGrid>
-        {products.map(({ node: product }, idx) => (
-          <UIProductGridItem key={`${product.id}`}>
-            <ProductCard.Component
-              aspectRatio={aspectRatio}
-              imgProps={{
-                width: 150,
-                height: 150,
-                sizes: '30vw',
-                loading: idx === 0 ? 'eager' : 'lazy',
-              }}
-              {...ProductCard.props}
-              bordered={bordered ?? ProductCard.props.bordered}
-              showDiscountBadge={
-                showDiscountBadge ?? ProductCard.props.showDiscountBadge
-              }
-              product={product}
-              index={pageSize * page + idx + 1}
-              taxesConfiguration={taxesConfiguration}
-            />
-          </UIProductGridItem>
-        ))}
+        {isFirstPage ? (
+          <>
+            {products.slice(0, 2).map(({ node: product }, idx) => (
+              <UIProductGridItem key={`${product.id}`}>
+                <ProductCard.Component
+                  aspectRatio={aspectRatio}
+                  imgProps={{
+                    width: 150,
+                    height: 150,
+                    sizes: '30vw',
+                    loading: 'eager',
+                  }}
+                  {...ProductCard.props}
+                  bordered={bordered ?? ProductCard.props.bordered}
+                  showDiscountBadge={
+                    showDiscountBadge ?? ProductCard.props.showDiscountBadge
+                  }
+                  product={product}
+                  index={pageSize * page + idx + 1}
+                  taxesConfiguration={taxesConfiguration}
+                />
+              </UIProductGridItem>
+            ))}
+            <></>
+            <ViewportObserver name="UIProductGrid-out-viewport">
```

Comment:
This is primarily for testing purposes at the moment. If you think it's more appropriate, we can include it in the next PR. We will have to apply `ViewportObserver` component in the _LazyLoadSection_.